### PR TITLE
Enable Steam VR text input

### PR DESCRIPTION
**Pull request by jherico**
Enables the SteamVR overlay keyboard when the HMD is active and a text input control is focused.  Needs polish and more design thought.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9883f0a8806fde42a3f425a6e7686c4175274f3c available at: https://gist.github.com/69c401c986002abe1877de0f2d2c9944<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7936.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7936.dmg

File: libraries/gl/src/gl/OffscreenQmlSurface.cpp [hunk 1]
```diff
@@ -742,3 +744,42 @@ QVariant OffscreenQmlSurface::returnFromUiThread(std::function<QVariant()> funct
...
+
+void OffscreenQmlSurface::onFocusObjectChanged(QObject* object) {
+    if (!object) {
+        setFocusText(false);
+        return;
+    }
+
+    QVariant result;
+#if 1
+    auto invokeResult = QMetaObject::invokeMethod(object, "inputMethodQuery", Q_RETURN_ARG(QVariant, result),
```
Comment (jherico) [inline libraries/gl/src/gl/OffscreenQmlSurface.cpp]: generates a warning if the new object does not support text input. need to figure out how to make the below code path work.

**Comment by hyperlogic** [comment]
It's kind of difficult to get the keyboard to appear.  I was only able to make it show once, then when I dismissed it, I could no longer get it to appear.

**Comment by hyperlogic** [comment]
One other interesting point, is that their keyboard seems to be more than 1 meter away, so when it overlaps with our own dialogs, it results in uncomfortable focus disparities.

**Comment by howard-stearns** [comment]
There's been a suggestion to alter our hud distance, to something like 1.5 meters. That could conceivably happen during the sprint.

(I'm a little hesitant about chasing that around too much. Oculus Home uses 2.5m, but they're not trying to make things "reachable" by an IK attached arm. DK-2 recommends 75cm, and Leap Motion recommends even less.)

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0ebd1c362736c87f8de07dc23ccec14e5752ef19 available at: https://gist.github.com/69c401c986002abe1877de0f2d2c9944<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7936.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7936.dmg

**Comment by jherico** [comment]
Moving to a new PR targeting the Vive UI branch.

**Comment by hifi-gustavo** [comment]
Build console output for commit: bedc2f3a90d93aff0a1dbf91c2265c80c5ba69f4 available at: https://gist.github.com/69c401c986002abe1877de0f2d2c9944<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7936.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7936.dmg

**Pull request closed without merging**